src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Debug/Designer/LaunchProfileEditorServiceFactory.cs

```diff
@@ -0,0 +1,93 @@
+using Microsoft.VisualStudio.RpcContracts;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.ServiceBroker;
+using Task = System.Threading.Tasks.Task;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Debug.Designer
+{
+    [Export(typeof(IPackageService))]
+    internal sealed class LaunchProfileEditorServiceFactory : IPackageService, IDisposable
+    {
+        private const string LaunchProfileEditorServiceName = "Microsoft.VisualStudio.ProjectSystem.Managed.LaunchProfileEditorService";
+        private const string ProjectGuidActivationArgumentName = "ProjectGuid";
+
+        internal static readonly ServiceJsonRpcDescriptor LaunchProfileEditorServiceDescriptorV1 = new(
+            new ServiceMoniker(LaunchProfileEditorServiceName, new Version(0, 1)),
+            ServiceJsonRpcDescriptor.Formatters.UTF8,
+            ServiceJsonRpcDescriptor.MessageDelimiters.HttpLikeHeaders);
+
+        private readonly IProjectServiceAccessor _projectServiceAccessor;
+        private IDisposable? _profferDisposable;
+        private ILaunchProfileEditorService? _launchProfileEditorService;
+
+        [ImportingConstructor]
+        public LaunchProfileEditorServiceFactory(IProjectServiceAccessor projectServiceAccessor)
+        {
+            _projectServiceAccessor = projectServiceAccessor;
+        }
+
+        public async Task InitializeAsync(IAsyncServiceProvider asyncServiceProvider)
+        {
+            IBrokeredServiceContainer brokeredServiceContainer = await asyncServiceProvider.GetServiceAsync<SVsBrokeredServiceContainer, IBrokeredServiceContainer>();
+
+            _profferDisposable = brokeredServiceContainer.Proffer(LaunchProfileEditorServiceDescriptorV1, CreateInstanceAsync);
+
+            async ValueTask<object?> CreateInstanceAsync(
+                ServiceMoniker moniker,
+                ServiceActivationOptions options,
+                IServiceBroker serviceBroker,
+                AuthorizationServiceClient authorizationServiceClient,
+                CancellationToken cancellationToken)
+            {
+                await authorizationServiceClient.AuthorizeOrThrowAsync(WellKnownProtectedOperations.CreateClientIsOwner(), cancellationToken);
+
+                if (options.ActivationArguments is null
+                    || !options.ActivationArguments.TryGetValue("ProjectGuid", out string projectGuidString))
+                {
+                    throw new InvalidOperationException($"Missing required activation argument \"{ProjectGuidActivationArgumentName}\".");
+                }
+
+                if (!Guid.TryParse(projectGuidString, out Guid projectGuid))
+                {
+                    throw new InvalidOperationException($"Unable to parse activation argument \"{ProjectGuidActivationArgumentName}\": \"{projectGuidString}\".");
+                }
+
+                if (options.ClientRpcTarget is not ILaunchProfileEditorClientSession clientSession)
```

Comment:
We also expect the client to provide an `ILaunchProfileEditorClientSession`; this is the interface we use when we want to communicate information back to the client (e.g., the list of launch profiles, updates to properties, etc.).